### DISCUSSION TITLE
Multidimensional timeseries

### DISCUSSION THREAD

**Discussion by AlexKrt** [ravendb/ravendb#12674] [Q&A]
Hi,
         I have 1024 x values and 1024 y values for every timestamp value. What would be the best way to save this as a timeseries?
There is a max 32 values per each timestamp. Any other way?
Thank you,

**Comment by ppekrol** [comment]
Hi @AlexKrt 

The only thing that comes to my mind would be to create separate TS'es with 32 values per each timestamp. Would that work for you?

**Comment by ayende** [comment]
@AlexKrt Can you explain what is the scenario where you have 1,024 values per time stamp?

The reason that we have multi value entries for time stamps is that you'll run into values that only make sense together.
For example, lat / lng for geo location are only meaningful together. 
Blood pressure value is the same.

In broader context, you may have a sensor suite that will report things like humidity, heat and wind direction and you want to capture that as a single unit.

However, the 32 value limit is imposed as a way to ensure that we can pack as much data into the system as possible. 

Having 1,024 values for a single time stamp would lead me to suspect that you are mixing different sensors at the same time.

**Reply by AlexKrt** [reply]
Hi @ayende, In chemistry spectra is a collection of absorbance vs wavelength. An instrument sends light into the reaction vessel at a slightly different wavelength and then reads the absorbance. It takes about 1/5th of a second to run a spectra. And it is typically ~1000 wavelength variations and therefore ~1000 absorbance readings. Spectra is like taking a picture of your reaction. That is why it has a single timestamp. A chemist may want to take a spectra once every couple seconds. A chemist may want to search for a particular spectra in time he sees a pick from other trend measurements. 
So spectra is an XY combinations at a particular time. Does it fit timeseries? How else to store it?

**Reply by ayende** [reply]
What do you *do* with those values?
If you have just a few, and aren't actually interested in doing value over time for individual values, there isn't much point.
You can store that as attachments directly.
